### ISSUE TITLE
Fix Request for false positive in AvoidExposingMutableRecordState

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#554]
```
import java.util.List;
import lombok.Builder;

@Builder
public record FooDto(
    List<SomeDto> someDates, // <-- false positive
    List<OtherDto> otherDates  // <-- false positive
){
    public FooDto {
        someDates = List.copyOf(someDates == null ? List.of() : someDates);
        otherDates = List.copyOf(otherDates == null ? List.of() : otherDates);
    }
}
```
